### PR TITLE
 dotCMS/core#15761 SPAEM: Create navigation from the DotCMS nav api 

### PR DESCRIPTION
**Pull request by fmontes**

### CODE REVIEW

File: src/__tests__/Nav.test.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    it('should have BootstrapNav and pill in true', () => {
+        expect(wrapper.find('Nav').props().pills).toBe(true);
+    });
+
+    it('should expand menu', async () => {
+        expect(wrapper.find('Collapse').props().isOpen).toBe(false);
+
+        wrapper.find('NavbarToggler').simulate('click');
+        await wrapper.update();
```
Comment (alfredo-dotcms) [inline src/__tests__/Nav.test.js]: is this the same as "fixture.detectChanges();" ?

File: src/__tests__/Nav.test.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        await wait();
```
Comment (alfredo-dotcms) [inline src/__tests__/Nav.test.js]: what this does?

File: src/__tests__/__snapshots__/Header.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (alfredo-dotcms) [inline src/__tests__/__snapshots__/Header.test.js.snap]: what is this file for?

File: src/Components/Nav.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    toggleOpen = e => {
+        this.setState({
+            ...this.state,
+            isOpen: !this.state.isOpen
+        });
+    };
+
+    render() {
+        return (
+            <>
```
[the remark this answers is not in the document]
Comment (fmontes) [reply src/Components/Nav.js]: https://reactjs.org/docs/fragments.html

File: src/__tests__/Nav.test.js (hunk 1 above)
Comment (fmontes) [reply src/__tests__/Nav.test.js]: Kinda, yes... https://airbnb.io/enzyme/docs/api/ShallowWrapper/update.html

File: src/__tests__/Nav.test.js (hunk 2 above)
Comment (fmontes) [reply src/__tests__/Nav.test.js]: Wait for the changes to propagate in the dom.

File: src/__tests__/__snapshots__/Header.test.js.snap (hunk 3 above)
Comment (fmontes) [reply src/__tests__/__snapshots__/Header.test.js.snap]: https://jestjs.io/docs/en/snapshot-testing

**Pull request merged**